### ISSUE TITLE
might create empty line ending with a solely comma

### ISSUE THREAD

**Issue by tsingakbar** [grailbio/bazel-compilation-database#60]
which again result in an invalid json, example:
```json
{"command":"xxxxxxxxxxxxxxx"},
,
{"command":"xxxxxxxxxxxxxx"},
```
introduced since tag0.4.4, still exists in 0.4.5.

temporarily add a sed command to delete these lines:
```shell
sed -i -e '/^,$/d' "${COMPDB_FILE}"
```

**Comment by tsingakbar** [comment]
This bug also introduced another one. If the last line is a solely comma, then the comma will be deleted, then the 2nd last line became the last line, and it ends with a comma, then the whole json is again not valid: last element in array ends with comma.

This kind of bug is endless, so why sticking to shell script? Use a scripting language with json support is not delicious enough?

**Comment by siddharthab** [comment]
Yes, I am thinking of rewriting in Python.

**Comment by tsingakbar** [comment]
https://github.com/tsingakbar/bazel-compilation-database/blob/master/generate.py

**Comment by siddharthab** [comment]
Should be resolved with ff7a98c3e.
